src/Explorer/Controls/Settings/SettingsSubComponents/MongoIndexingPolicy/MongoIndexingPolicyComponent.tsx

```diff
@@ -0,0 +1,406 @@
+          <>
+            <DetailsList
+              styles={customDetailsListStyles}
+              disableSelectionZone
+              items={initialIndexes}
+              columns={this.initialIndexesColumns}
+              selectionMode={SelectionMode.none}
+              onRenderRow={this.onRenderRow}
+              layoutMode={DetailsListLayoutMode.justified}
+            />
+            {this.renderIndexesToBeAdded()}
+          </>
+        )}
+      </Stack>
+    );
+  };
+
+  private renderIndexesToBeDropped = (): JSX.Element => {
+    const indexesToBeDropped: MongoIndexDisplayProps[] = this.props.indexesToDrop.map(
+      (dropIndex: number, arrayPosition: number) =>
+        this.getMongoIndexDisplayProps(this.props.mongoIndexes[dropIndex], arrayPosition, false)
+    );
+
+    return (
+      <Stack styles={mediumWidthStackStyles}>
+        <Stack
+          className="collapsibleLabel"
+          horizontal
+          tokens={accordionStackTokens}
+          onClick={this.toggleIndexesToBeDroppedExpanded}
+        >
+          <Icon
+            iconName={this.state.indexesToBeDroppedExpanded ? "ChevronDown" : "ChevronRight"}
+            styles={accordionIconStyles}
+          />
+          <Label>Index(es) to be dropped</Label>
+        </Stack>
+        {this.state.indexesToBeDroppedExpanded && indexesToBeDropped.length > 0 && (
+          <DetailsList
+            styles={customDetailsListStyles}
+            disableSelectionZone
+            items={indexesToBeDropped}
+            columns={this.indexesToBeDroppedColumns}
+            selectionMode={SelectionMode.none}
+            onRenderRow={this.onRenderRow}
+            layoutMode={DetailsListLayoutMode.justified}
+          />
+        )}
+      </Stack>
+    );
+  };
+
+  private refreshIndexTransformationProgress = async () => {
+    this.setState({ isRefreshingIndexTransformationProgress: true });
+    try {
+      await this.props.refreshIndexTransformationProgress();
+    } finally {
+      this.setState({ isRefreshingIndexTransformationProgress: false });
+    }
+  };
+
+  public isIndexingTransforming = (): boolean =>
+    this.props.indexTransformationProgress !== undefined && this.props.indexTransformationProgress !== 100;
+
+  private onClickRefreshIndexingTransformationLink = async () => await this.refreshIndexTransformationProgress();
+
+  private renderIndexTransformationWarning = (): JSX.Element => {
+    let indexTransformationWarning: JSX.Element;
+    if (this.state.isRefreshingIndexTransformationProgress) {
+      indexTransformationWarning = (
+        <Stack horizontal {...mongoWarningStackProps}>
+          <Text>Refreshing indexing policy update information</Text>
+          <Spinner size={SpinnerSize.medium} />
+        </Stack>
+      );
+
+      // index transformation progress can be 0
+    } else if (this.isIndexingTransforming()) {
+      const updateInfoString = "The indexing policy is being updated in the background. ";
```

Comment:
Consider updating the `NotificationConsole` (it has a `Progress` type, so it can display transient messages).

Reply:
We need the temp notification with spinner to show up here on top